python/paddle/v2/fluid/framework.py

```diff
@@ -741,3 +742,40 @@ def program_guard(main_program, startup_program=None):
     switch_main_program(main_program)
     if startup_program is not None:
         switch_startup_program(startup_program)
+
+
+_device_stack_ = []
+
+
+class Device(object):
+    def __init__(self, device):
+        self.device = device
+
+    def __enter__(self):
+        _device_stack_.append(self.device)
+
+    def __exit__(self, type, value, traceback):
+        _device_stack_.pop()
+
+
+_start_op_device_vec_ = []
+_main_op_device_vec_ = []
```

Comment:
What is `start_op_device_vec`, `start_op_device_vec`? the device lists of operators? Need to add comments.